sklearn/utils/_set_output.py

```diff
@@ -33,7 +33,7 @@ def get_columns(columns):
 class ContainerAdapterProtocol(Protocol):
     container_lib: str
 
-    def create_container(self, X_output, X_original, columns):
+    def create_container(self, X_output, X_original, columns, copy=False):
```

Comment:
I find `copy` quite odd here. It's not that we copy the data, which is the main part. We just make sure to create a new dataframe. Maybe `in_place`?